spark_pipeline_framework/transformers/fhir_sender/v1/fhir_sender.py

```diff
@@ -364,6 +390,33 @@ def _transform(self, df: DataFrame) -> DataFrame:
                 desired_partitions: int = num_partitions or math.ceil(
                     row_count / batch_size
                 )
+                if partition_by_column_name:
+                    json_df = json_df.withColumn(
+                        partition_by_column_name,
+                        get_json_object(col("value"), f"$.{partition_by_column_name}"),
+                    ).repartition(desired_partitions, partition_by_column_name)
+                    json_df = json_df.drop(partition_by_column_name)
```

Comment:
should be moved in else of `if run_synchronously:`

Reply:
In that case, all the three argument conditions will be moved to else condition because in order to make all the 3 arguments work, we need to define them in an order. Repartition -> sort within partitions -> drop fields

In case we're moving all  to else condition, then drop fields functionality would not work in the case of run_synchronously.